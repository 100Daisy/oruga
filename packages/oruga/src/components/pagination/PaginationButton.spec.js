import { shallowMount } from '@vue/test-utils'
import OPaginationButton from '@components/pagination/PaginationButton'

let wrapper
const page = 5
const defaultProps = {
    page: {
        number: page,
        isCurrent: false,
        click: jest.fn(),
        disabled: false,
        class: '',
        'aria-label': ''
    }
}

describe('OPagination', () => {
    beforeEach(() => {
        wrapper = shallowMount(OPaginationButton, {
            propsData: defaultProps
        })
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('OPaginationButton')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('returns href as expected', () => {
        expect(wrapper.vm.href).toBe('#')

        wrapper.setProps({ tag: 'button' })
        expect(wrapper.vm.href).toBeUndefined()
    })
})
