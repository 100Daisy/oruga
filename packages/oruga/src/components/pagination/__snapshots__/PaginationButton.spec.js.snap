// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OPagination render correctly 1`] = `<a role="button" href="#" aria-label="" class="">5</a>`;
